src/pkg/packager/lint/lint.go

```diff
@@ -50,16 +64,194 @@ func ValidateZarfSchema(path string) (*Validator, error) {
+		})
+	}
+	templateMap := map[string]string{}
+
+	setVarsAndWarn := func(templatePrefix string, deprecated bool) {
+		yamlTemplates, err := utils.FindYamlTemplates(node, templatePrefix, "###")
+		if err != nil {
+			validator.addWarning(validatorMessage{
+				description:    err.Error(),
+				packageRelPath: node.GetRelativeToHead(),
+				packageName:    node.GetOriginalPackageName(),
+			})
+		}
+
+		for key := range yamlTemplates {
+			if deprecated {
+				validator.addWarning(validatorMessage{
+					description:    fmt.Sprintf(lang.PkgValidateTemplateDeprecation, key, key, key),
+					packageRelPath: node.GetRelativeToHead(),
+					packageName:    node.GetOriginalPackageName(),
+				})
+			}
+			_, present := createOpts.SetVariables[key]
+			if !present {
+				validator.addWarning(validatorMessage{
+					description:    lang.UnsetVarLintWarning,
+					packageRelPath: node.GetRelativeToHead(),
+					packageName:    node.GetOriginalPackageName(),
+				})
+			}
+		}
+		for key, value := range createOpts.SetVariables {
+			templateMap[fmt.Sprintf("%s%s###", templatePrefix, key)] = value
+		}
+	}
+
+	setVarsAndWarn(types.ZarfPackageTemplatePrefix, false)
+
+	// [DEPRECATION] Set the Package Variable syntax as well for backward compatibility
+	setVarsAndWarn(types.ZarfPackageVariablePrefix, true)
 
+	utils.ReloadYamlTemplate(node, templateMap)
+}
+
+func isPinnedImage(image string) (bool, error) {
+	transformedImage, err := transform.ParseImageRef(image)
+	if err != nil {
+		if strings.Contains(image, types.ZarfPackageTemplatePrefix) ||
+			strings.Contains(image, types.ZarfPackageVariablePrefix) {
+			return true, nil
+		}
+		return false, err
+	}
+	return (transformedImage.Digest != ""), err
+}
+
+func isPinnedRepo(repo string) bool {
+	return (strings.Contains(repo, "@"))
+}
+
+func lintComponent(validator *Validator, node *composer.Node) {
+	checkForUnpinnedRepos(validator, node)
+	checkForUnpinnedImages(validator, node)
+	checkForUnpinnedFiles(validator, node)
+}
+
+func checkForUnpinnedRepos(validator *Validator, node *composer.Node) {
+	for j, repo := range node.Repos {
+		repoYqPath := fmt.Sprintf(".components.[%d].repos.[%d]", node.GetIndex(), j)
+		if !isPinnedRepo(repo) {
+			validator.addWarning(validatorMessage{
+				yqPath:         repoYqPath,
+				packageRelPath: node.GetRelativeToHead(),
+				packageName:    node.GetOriginalPackageName(),
+				description:    "Unpinned repository",
+				item:           repo,
+			})
+		}
+	}
+}
+
+func checkForUnpinnedImages(validator *Validator, node *composer.Node) {
+	for j, image := range node.Images {
+		imageYqPath := fmt.Sprintf(".components.[%d].images.[%d]", node.GetIndex(), j)
+		pinnedImage, err := isPinnedImage(image)
+		if err != nil {
+			validator.addError(validatorMessage{
+				yqPath:         imageYqPath,
+				packageRelPath: node.GetRelativeToHead(),
+				packageName:    node.GetOriginalPackageName(),
+				description:    "Invalid image format",
```

Comment:
```suggestion
				description:    "Invalid image reference",
```
Format might imply something like its format on disk